python/ray/data/_internal/execution/backpressure_policy.py

```diff
@@ -0,0 +1,100 @@
+import logging
+import os
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ray.data._internal.execution.interfaces.physical_operator import (
+        PhysicalOperator,
+    )
+    from ray.data._internal.execution.streaming_executor_state import Topology
+
+logger = logging.getLogger(__name__)
+
+
+# TODO(hchen): Enable ConcurrencyCapBackpressurePolicy by default.
+DEFAULT_BACKPRESSURE_PLOCIES = []
```

Comment:
So do we foresee in the near future, we will have multiple policies enabled at same time?

Reply:
I do plan to migrate existing backpressure code to this interface. 
Another reason why I wanted to introduce this interface is that it will allow us experimenting new back-pressure policies without touching the code base. E.g., one idea is to take the real runtime metrics into consideration for backpressure.

Reply:
Sounds good, I don't feel strongly about it, just think that it may be too early to abstract (we don't know yet if this is the right interface).

Reply:
yes, the interface is likely to change as we implement other policies. it's an internal interface, so that should be fine.